### PR TITLE
add npairs loss

### PR DESCRIPTION
**Pull request by kebinC**

### CODE REVIEW

File: fluid/PaddleCV/metric_learning/losses/npairsloss.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        label_remapped = fluid.layers.equal(anchor_lab, positive_lab_tran)
+        label_remapped = fluid.layers.cast(label_remapped, dtype='float32') / (samples_each_class-1)
+        label_remapped.stop_gradient = True
+
+        out = fluid.layers.softmax(input=similarity_matrix, use_cudnn=False)
+        xentloss = fluid.layers.cross_entropy(input=out, label=label_remapped, soft_label=True)
+        xentloss = fluid.layers.mean(x=xentloss)
+
+        reg = fluid.layers.reduce_mean(fluid.layers.reduce_sum(fluid.layers.square(input), dim=1))
+        l2loss = 0.5 * reg_lambda * reg
```
Comment (qingqing01) [inline fluid/PaddleCV/metric_learning/losses/npairsloss.py:57]: 这里面npair loss是否要换成Paddle提供的API?

Comment (kebinC) [reply fluid/PaddleCV/metric_learning/losses/npairsloss.py:57]: 1) Paddle提供的npair loss可能在这个代码结构里还会有些问题， 我看了里面用了softmax_with_cross_entropy, 这个会因为形状是-1的问题导致跑不起来
2) 目前loss里面支持一个anchor对应多个positive的情况，这个没办法调用api

**Review by qingqing01 — APPROVED** [review]
